test/HooglePlus/FilterTestSpec.hs

```diff
@@ -0,0 +1,54 @@
+module HooglePlus.FilterTestSpec (spec) where
+
+
+import Test.Hspec
+import Text.Pretty.Simple
+import Text.Parsec.Pos
+import Control.Monad.State
+import Text.Parsec.Indent
+import qualified Data.Map as Map
+import Data.Either
+
+import HooglePlus.FilterTest
+
+importList = ["Prelude", "Data.List", "Data.Maybe"]
+
+spec :: Spec
+spec =
+  describe "Filter" $ do
+
+  it "Return Correct Type" $ do
+    result <- getTypeString importList "\\x -> [x]"
+    case result of
+      Left err -> error "failed"
+      Right str -> str `shouldBe` "a -> [a]"
+  
+  it "Succeed on poly function w/o type constrains" $ do
+    result <- runNoExceptionTest "\\x -> x"
+    result `shouldBe` True
+
+  it "Pass function w/ different retval - singleList" $ do
+    result <- runGhcChecks "\\x -> [x]"
+    result `shouldBe` True
+
+  it "Pass function w/ different retval - headLast" $ do
+    result <- runGhcChecks "\\x -> (head x, last x)"
+    result `shouldBe` True
+
+  it "Fail on invalid fcns - head []" $ do
+    result <- runGhcChecks "\\x -> (head x, last [])"
+    result `shouldBe` False
+
+  it "Fail on invalid fcns - just nothing" $ do
+    result <- runGhcChecks "\\x -> fromJust Nothing"
+    result `shouldBe` False
+
+  it "Fail on const fcns" $ do
+    result <- runGhcChecks "\\x -> \\y -> const x y"
+    result `shouldBe` False
```

Comment:
Maybe you should make this a mapM over pairs of input-output pairs!

Also could you add this case?
`("\\arg0 -> Data.ByteString.Builder.toLazyByteString (Data.ByteString.Builder.int64Dec arg0)", True)`

The type of the function is `GHC.Int.Int64 -> Data.ByteString.Lazy.ByteString`. Can we handle `Int64`s? If not, that's okay for now.

I'm concerned that the module names aren't part of input, when they would be, iirc.

Reply:
Also, why does this call `runGhcChecks` instead of your specific `runCheck` or `progDoesNotCrash` function?

Reply:
The spec file was outdated and I'm working on the new one, should be done in ~5min